src/utils/mod.rs

```diff
@@ -0,0 +1,69 @@
+use std::{fs::{self, File}, io::Write};
+
+use reqwest::{blocking::Client, header::USER_AGENT, blocking::Response};
+use serde::Deserialize;
+use sevenz_rust2::Archive;
+
+pub fn download_database_from_github() -> Result<(),reqwest::Error> {
```

Comment:
Something to maybe think about are these methods that return a Result, but also have a bunch  of panic paths:

There's a crate called `anyhow` which Has a Result<T>, where T is the Ok type.  You do not have to define the error type. This allows you to convert those `.unwrap()` to a simple `?`. This would remove most of your current panic paths.